test/services/users.test.ts

```diff
@@ -1,13 +1,245 @@
+      userId = id;
+    });
+    after(async () => {
+      await service.remove(userId);
+    });
+
+    it('should succeed - get user by id', async () => {
+      const { email, id } = await service.get(userId);
+      expect(id).to.exist;
+      expect(email).to.equal(userInfo.email);
+    });
+
+    it('should fail - not existed userId', async () => {
+      try{
+        await service.get(unexistedId);
+        expect.fail('call should have failed');
+      } catch (err: any) {
+        const {code} = err;
+        expect(code).to.equal(404);
+      }
+    });
+
+    it('should fail - invalid userId', async () => {
+      try{
+        await service.get('123456');
+        expect.fail('call should have failed');
+      } catch (err: any) {
+        const {code, message} = err;
+        expect(code).to.equal(400);
+        expect(message).to.equal('ValidationError: "id" must be a valid GUID');
+      }
+    });
+  });
+
+  describe('find user', () => {
+
+    before(async () => {
+      const {id} = await service.create(userInfo);
+      userId = id;
+    });
+    after(async () => {
+      await service.remove(userId);
+    });
+
+    it('should succeed - find user', async () => {
+      const {total} = await service.find();
+      expect(total).not.to.equal(0);
+    });
+  });
+
+  describe('update (patch) user', () => {
+
+    before(async () => {
+      const {id} = await service.create(userInfo);
+      userId = id;
+    });
+    after(async () => {
+      await service.remove(userId);
+    });
+
+    it('should succeed - update(patch) user', async () => {
+      const { firstName, lastName } = await service.patch(userId, updateUser);
+      expect(firstName).to.equal(updateUser.firstName);
+      expect(lastName).to.equal(updateUser.lastName);
+    });
+
+    it('should succeed - user was updated(patch) successfully', async () => {
+      const { firstName, lastName } = await service.get(userId);
+      expect(firstName).to.equal(updateUser.firstName);
+      expect(lastName).to.equal(updateUser.lastName);
+    });
+
+    it('should fail - invalid user id', async () => {
+      try{
+        await service.patch(unexistedId,updateUser);
+        expect.fail('call should have failed');
+      } catch (err: any) {
+        const {code} = err;
+        expect(code).to.equal(404);
+      }
+    });
+
+    it('should fail - invalid user data', async () => {
+      const invalidUserData ={
+        firstName: 12,
+        lastName: 'updated user last name',
+        email: 123456,
+        password: 'supersecret'
+      };
+      try{
+        await service.patch(userId,invalidUserData);
+        expect.fail('call should have failed');
+      } catch (err: any) {
+        const {code, message} = err;
+        expect(code).to.equal(400);
+        expect(message).to.equal('Invalid data');
+      }
+    });
+  });
+
+  describe('delete user', () => {
+
+    it('should fail - id not found', async () => {
```

Comment:
this test is not needed. only disallow test

Reply:
Only "not allowed =405" test is remaining for delete ed2fcab9e7527ff7434ea9879bb56714970dc4f7